### PR TITLE
chore: simplify types for components with static properties

### PR DESCRIPTION
**Pull request by havenchyk**
[FX-NNNN]

### Description

@teimurjan found a cool approach (I saw a couple of times) to have proper typings of static props on compound components with no explicit type coercing. 

To me it makes sense to simplify our types and apply Teimur's idea for whole the codebase

### How to test

- FIXME: Add the steps describing how to verify your changes

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

[FX-1969]: https://toptal-core.atlassian.net/browse/FX-1969

### CODE REVIEW

File: packages/picasso-provider/src/Picasso/story/DisableResponsiveUI.example.tsx [hunk 1]
```diff
@@ -28,15 +28,7 @@ const Example = () => (
...
-const SidebarMenu = () => (
-  <Sidebar>
-    <Sidebar.Menu>
-      <Sidebar.Item>Home</Sidebar.Item>
-      <Sidebar.Item>Contacts</Sidebar.Item>
-      <Sidebar.Item>Team</Sidebar.Item>
-    </Sidebar.Menu>
-  </Sidebar>
-)
+const SidebarMenu = () => <Sidebar>klahsdlfkdajsf</Sidebar>
```
Comment (TomasSlama) [inline packages/picasso-provider/src/Picasso/story/DisableResponsiveUI.example.tsx:31]: was this intention?

**Review by TomasSlama — APPROVED** [review]

File: packages/picasso-provider/src/Picasso/story/DisableResponsiveUI.example.tsx (hunk 1 above)
Comment (teimurjan) [reply packages/picasso-provider/src/Picasso/story/DisableResponsiveUI.example.tsx:31]: I don't think so :)

File: packages/picasso/src/Avatar/Avatar.tsx [hunk 2]
```diff
@@ -112,6 +72,37 @@ export const Avatar: FunctionComponent<Props> = props => {
 
   const InputComponent = isBrowserSupportsObjectFit ? Image : IE11Image
 
+  const renderLogo = () => {
```
Comment (teimurjan) [inline packages/picasso/src/Avatar/Avatar.tsx:75]: It looks like this change is not about simplifying the types. Don't you want to extract it to a separate PR?

File: packages/picasso/src/Button/Button.tsx [hunk 3]
```diff
@@ -230,10 +224,11 @@ Button.defaultProps = {
...
-Button.Group = Group
-Button.Circular = Circular
-Button.Action = Action
-
-export default Button as OverridableComponent<Props> & StaticProps
+export default Object.assign(Button, {
+  Group,
+  Circular,
+  Action
+  // Split: Split
```
Comment (teimurjan) [inline packages/picasso/src/Button/Button.tsx:233]: Why is it here?

File: packages/picasso/src/Button/Button.tsx [hunk 4]
```diff
@@ -25,6 +24,7 @@ import Container from '../Container'
 import Group from '../ButtonGroup'
 import Circular from '../ButtonCircular'
 import Action from '../ButtonAction'
+// import Split from '../ButtonSplit'
```
Comment (teimurjan) [inline packages/picasso/src/Button/Button.tsx:27]: Why is it here?

File: packages/picasso/src/Pagination/Pagination.tsx [hunk 5]
```diff
@@ -5,7 +5,7 @@ import React, {
   HTMLAttributes
```
Comment (teimurjan) [inline packages/picasso/src/Pagination/Pagination.tsx:5]: I'd not change this file because it can cause conflicts with https://github.com/toptal/picasso/pull/2155.

File: packages/picasso-lab/src/Note/Note.tsx [hunk 6]
```diff
@@ -31,13 +25,14 @@ export const Note = forwardRef<HTMLDivElement, Props>(function Note (
...
-Note.Subtitle = NoteSubtitle
-Note.Content = NoteContent
+Note.displayName = 'PicassoNote'
 
-export default Note
+export default Object.assign(Note, {
+  Title: NoteTitle,
+  Subtitle: NoteSubtitle,
+  Content: NoteContent
+})
```
Comment (denieler) [inline packages/picasso-lab/src/Note/Note.tsx:38]: cool approach 👍 interesting, why it works, but just assignment of props doesn't 🤔

Comment (teimurjan) [reply packages/picasso-lab/src/Note/Note.tsx:38]: Because `assign` function has correct typing. I've discovered it when fixing a bug for a nested component with generic props.

File: packages/picasso/src/Button/Button.tsx (hunk 3 above)
Comment (havenchyk) [reply packages/picasso/src/Button/Button.tsx:233]: because button split is not merged yet, wrong change, thanks

File: packages/picasso-lab/src/Note/Note.tsx (hunk 6 above)
Comment (havenchyk) [reply packages/picasso-lab/src/Note/Note.tsx:38]: this is findings of Teimur, TypeScript can detect types correctly

Comment (havenchyk) [reply packages/picasso-lab/src/Note/Note.tsx:36]: `assign<T, U>(target: T, source: U): T & U;`
